### PR TITLE
Product filters (Rating filter and Optimization of filter functionality)

### PR DESCRIPTION
**Pull request by saicharan-gouru**
-Added rating filter functionality
-Added filter functionality on click of categories in home page
-Clear filter functionality on click of product listing page
Spazeir-mart-productfilter2

### CODE REVIEW

File: src/pages/homepage/components/Categories.js [hunk 1]
```diff
@@ -2,18 +2,30 @@ import "./Categories.css";
...
+
+    function generateType(categoryName){
+        if(categoryName === "Staples")
+        return "INCLUDE_STAPLES"
+        if(categoryName === "Snacks")
+        return "INCLUDE_SNACKS"
+        if(categoryName === "Dairy&Eggs")
+        return "INCLUDE_DAIRY_AND_EGGS"
+        if(categoryName === "Vegetables")
+        return "INCLUDE_VEGETABLES"
```
Comment (maitrakhatri) [inline src/pages/homepage/components/Categories.js:18]: just a suggestion: can use `switch` here

**Review by maitrakhatri — APPROVED** [review]
Looks is very clean and easy to read.

File: src/pages/productlistingpage/components/Filters.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import {useData} from "../../../index";
 
 function Filters(){
 
-    const {sortByPrice,productsDispatch,includeStaples,includeSnacks,includeDairyandEggs,includeVegetables} = useData();
+    const {sortByPrice,productsDispatch,includeStaples,includeSnacks,includeDairyandEggs,includeVegetables,rating} = useData();
     console.log(sortByPrice)
```
Comment (Tanay-J) [inline src/pages/productlistingpage/components/Filters.js:7]: You may remove unwanted code before merging

**Review by Tanay-J — APPROVED** [review]
Most of the code has been reviewed in other PR. Current changes look good to merge. Please consider using ```switch``` as suggested by other reviewer. Approving the changes.

**Pull request merged**